File: minwook/8주차/푸드_파이트_대회.java
class Solution {
    public String solution(int[] food) {
        String answer = "";
        int f = 0;
        for(int i=1; i<food.length; i++){
            f = food[i]/2;
            if(f >= 1){
                while(f>0){
                    answer += i;
                    f--;
                }
            }
        }
        answer += '0';
        for(int i=food.length-1; i>0; i--){
            f = food[i]/2;
            if(f >= 1){
                while(f>0){
                    answer += i;
                    f--;
                }
            }
        }
        return answer;
    }
    
} 
// 양쪽에 같은 숫자를 배치하고 가운데 0을 삽입해야함
// 앞뒤로 숫자 추가 or StringBuilder 사용하면 더 좋은 코드가 가능할듯

File: yongkeun/8week/푸드파이트대회.js
function solution(food) {
  let leftPlayer = "";
  for (let i = 1; i < food.length; i++) {
    leftPlayer += i.toString().repeat(parseInt(food[i] / 2)); //toString으로 i를 문자로 바꾸고 repeat 안에 음식의 양을 2개로 나눈 값의 몫만 가져와 answer에는 1113과 같은 값이 들어가도록 함
  }
  rightPlayer = leftPlayer.split("").reverse().join("");
  //그 후에 1113을 다 가져왔으면 0(물)을 넣고 1113을 거꾸로 뒤집은 3111을 뒤집어서 답을 완성함
  return `${leftPlayer}0${rightPlayer}`;
}
